src/java/com/puppetlabs/http/client/impl/JavaClient.java

```diff
@@ -324,27 +337,52 @@ public static void requestWithClient(final RequestOptions requestOptions,
 
         final HttpContext httpContext = HttpClientContext.create();
 
-        final FutureCallback<HttpResponse> futureCallback = new FutureCallback<HttpResponse>() {
+        final Timer.Context timerContext = registry == null ? null : timer(registry, request).time();
```

Comment:
This need to start the timer outside the ```TimedFutureCallback``` but stop it inside seems slightly funny to me. I wonder would it be nice to make ```TimedFutureCallback``` a delegate that takes a registry, request and callback as constructor args? We could add a ```start()``` method too that could be called closer to the moment Apache's ```execute()``` method is called. That might remove the need for the separate ```timedXxxx()``` methods too.